providers/universal-provider/src/utils/deepLinks.ts

```diff
@@ -0,0 +1,31 @@
+import { SignClient } from "@walletconnect/sign-client";
+import { SignClientTypes } from "@walletconnect/types";
+import { RequestParams } from "../types";
+
+export const handleDeepLinks = (client: InstanceType<typeof SignClient>, params: RequestParams) => {
+  const handleSessionRequestSent = (
+    payload: SignClientTypes.EventArguments["session_request_sent"],
+  ) => {
+    // only handle the request if it matches the request and topic
+    if (payload.request !== params.request || payload.topic !== params.topic) return;
+    client.events.removeListener("session_request_sent", handleSessionRequestSent);
+    deeplinkRedirect();
+  };
+  client.on("session_request_sent", handleSessionRequestSent);
```

Comment:
Maybe we can use `.once` here instead of `.on`

Reply:
`.once` will not work if a different request is received by the listener. In the fx, after confirming we've received the correct address, I'm removing the listener so we shouldn't have issues with memory leaks